src/aragonjs-wrapper.js

```diff
@@ -41,26 +169,36 @@ const initWrapper = async (
     onAccounts = noop,
   } = {}
 ) => {
+  const isDomain = /[a-z0-9]+\.aragonid\.eth/.test(dao)
+
+  const daoAddress = isDomain
+    ? await resolveEnsDomain(dao, {
+        provider,
+        registryAddress: ensRegistryAddress,
+      })
+    : dao
+
+  if (!daoAddress) {
+    console.log('Invalid DAO address')
+    return
+  }
+
   const wrapper = new Aragon(daoAddress, {
     ensRegistryAddress,
     provider,
     apm: { ipfs: ipfsConf },
   })
 
-  // TODO: using window.web3 instead of new Web3(provider) for now, to make
-  // MetaMask work with Ganache.
-
-  // const web3 = new Web3(walletProvider || provider)
-  const web3 = window.web3
+  const web3 = new Web3(walletProvider || provider)
   onWeb3(web3)
 
-  const pollAccounts = () => {
-    web3.eth.getAccounts((err, accounts) => {
-      onAccounts(accounts || [])
-      setTimeout(pollAccounts, ACCOUNTS_POLL_EVERY)
-    })
+  const account = await getMainAccount(web3)
+  if (account === null) {
+    throw new Error(
+      'No accounts detected in the environment (try to unlock your wallet)'
+    )
   }
-  pollAccounts()
+  onAccounts([account])
```

Comment:
Do we still need this part here to get the account, given that the `<App />` seems to already do this in it's `componentDidMount()`?

Reply:
Yes let’s remove this